src/test/java/com/kodilla/ecommercee/ProductTests.java

```diff
@@ -0,0 +1,136 @@
+package com.kodilla.ecommercee;
+
+import com.kodilla.ecommercee.domain.Product;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+
+@Transactional
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class ProductTests {
+
+    @Autowired
+    private GenericEntityRepository genericEntityRepository;
+
+
+    @Test
+    public void testSaveProduct() {
+
+        //Given
+        Product product = new Product("Altana SunSet", "Altana ogrodowa ...", 999.90);
+
+        //When
+        genericEntityRepository.save(product);
+
+        //Then
+        Long id = product.getId();
+        Product foundProduct = (Product) genericEntityRepository.getOne(id);
+        Assert.assertNotEquals(null, foundProduct);
+
+        //CleanUp
+        genericEntityRepository.deleteById(id);
```

Comment:
Sekcja clean up wydaje mi się,że powinna wygllądać na tej zasadzie jak w module 17(submoduł 17.2,podpunkt sprawdzanie-testy kontrollera repozytorium): (w sensie żeby wskazać konkretnie,że taska o id 0,(bo jednego utworzylismy),chcemy skasować z bazy
//CleanUp
        //int id = readTasks.get(0).getId();
        //taskDao.deleteById(id);

Reply:
@aleksandra4991 W JDP190502-18 Paweł napisał, że @Transactional powoduje, że dane testowe zostaną usunięte, wiec usune CleanUp we wszystkich testach.